plugins/grype/copa-grype.go

```diff
@@ -0,0 +1,62 @@
+package main
```

Comment:
this would require all plugins to be in-tree, is that right?

Reply:
right, like the following:

plugins
├── grype
│   └── copa-grype.go
├── plugin2
│   └── copa-plugin2.go
└── ...

Reply:
Nope, not necessary for it to be in-tree. We can have a separate repo for each plugin just like in os exec approach.
For example: [https://github.com/anubhav06/copa-grype](https://github.com/anubhav06/copa-grype). This repo contains code for building a binary which can be run using the os exec approach. Instead, we can have the code for building the `.so` binary which can be run using the go plugins (this) approach.

However, now I'm thinking that we cannot add a `.so` file to our $PATH, right? Unlike os exec approach where the binary can be added to the path and can be automatically invoked; here we will have to specify the path of the `.so` file when running copa (--scanner <path_to_so-plugin_file> ). So this might not be a good approach

Reply:
If we specify a directory where the plugin files need to be, that could work. Like `~/.copa/plugins`.